src/cache/cache.rs

```diff
@@ -87,15 +85,19 @@ pub struct CacheRead {
 impl CacheRead {
     /// Create a cache entry from `reader`.
     pub fn from<R: ReadSeek + 'static>(reader: R) -> io::Result<CacheRead> {
```

Comment:
We could just make this return `error::Result<CacheRead>` and add `zip::result::ZipResult` to the error-chain.